app.py

```diff
@@ -11,700 +11,304 @@
+# done storing custom parameters
+########
 
 
 @app.callback(
-    Output("submission_indicator", "value"),
-    Output("submission_timer", "interval"),
-    Output("submission_timer", "disabled"),
-    Output("tabs", "active_tab", allow_duplicate=True),
-    Input("btn_solve_cqm", "n_clicks"),
-    Input("submission_timer", "n_intervals"),
-    State("submission_indicator", "value"),
-    State("tabs", "active_tab"),
-    State("built-sched", "children"),
-    prevent_initial_call=True
+    Output("availability-content", "children"),
+    Output("schedule-content", "children", allow_duplicate=True),
+    Output("schedule-tab", "disabled", allow_duplicate=True),
+    Output("tabs", "value"),
+    inputs=[Input("num-employees-select", "value"), Input("seed-select", "value")],
 )
-def submission_mngr(
-    n_clicks,
-    n_intervals,
-    submission_indicator_val,
-    active_tab,
-    built_sched
-):
-    trigger = callback_context.triggered
-    trigger_id = trigger[0]["prop_id"].split(".")[0]
-
-    if trigger_id == "btn_solve_cqm" and n_clicks:
-        return "Submitting... please wait", 1000, False, "avail"     
-        
-    if trigger_id == "submission_timer" and submission_indicator_val == "Submitting... please wait":
-
-        # Checks all current schedule values
-        s = set( val[0] for dic in built_sched['props']['derived_virtual_data'] for val in dic.values())
-        
-        if " " not in s: # Scheduled blocks are blank in final schedule
-            return no_update, no_update, False, no_update
-        else:
-            return "Done", no_update, True, "sched"
-        
-    return no_update, no_update, True, no_update
-
-   
-@app.callback(
-    Output("built-sched", "children", allow_duplicate=True),
-    Output("error_card_body", "children"),
-    Output("tabs", "active_tab"),
-    Input("submission_indicator", "value"),
-    State("shifts-per-employee-slider", "value"),
-    State("employees-per-shift-slider", "value"),
-    State("checklist-input", "value"),
-    State("cons shifts", "value"),  # consecutive shifts allowed
-    State("initial-sched", "children"),
-    State("built-sched", "children"),
-    State("error_card_body", "children"),
-    prevent_initial_call=True
+def disp_initial_sched(
+    num_employees: int, rand_seed: int
+) -> tuple[DataFrame, DataFrame, bool, str]:
+    """Display initial availability schedule.
+
+    Display initial schedule in, and switch to, the availability
+    tab if number of employees or seed is changed.
+    """
+    # one less to account for trainee
+    num_employees -= 1
+
+    shifts = [str(i + 1) for i in range(NUM_DAYS_IN_MONTH)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
+
+    init_availability_table = utils.display_availability(df, NOW.month, NOW.year)
+    return (
+        init_availability_table,
+        init_availability_table,
```

Comment:
I noticed that before this function was returning two different things here, what's the reasoning for having them be the same?

Reply:
It displays the availability table in the schedule tab for when the results are loading (so the greyed out table while running). The tab is disabled in this same return and only enabled when the schedule tab is loading and after the results are displayed, so you should never see the availability table in the schedule tab except when greyed out during loading.

Reply:
Oh got it, that makes sense, thanks for clarifying!